platform-common-pdf/src/main/java/com/softicar/platform/common/pdf/PdfPageRemover.java

```diff
@@ -0,0 +1,79 @@
+package com.softicar.platform.common.pdf;
+
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.pdfbox.pdmodel.PDDocument;
+
+/**
+ * Facilitates removing pages from PDF documents.
+ *
+ * @author Alexander Schmidt
+ */
+public class PdfPageRemover {
+
+	private static final int FAST_RENDERING_DPI = 10;
+
+	/**
+	 * Identifies all consecutive leading blank pages in the given PDF byte
+	 * array, and returns a new PDF byte array in which those leading blank
+	 * pages are omitted.
+	 *
+	 * @param pdfBytes
+	 *            the original PDF, as a byte array (never <i>null</i>)
+	 * @return the resulting PDF, as a byte array (never <i>null</i>)
+	 */
+	public byte[] removeLeadingBlankPages(byte[] pdfBytes) {
+
+		try (var document = PDDocument.load(pdfBytes)) {
+			determineBlankPageIndexes(pdfBytes)//
+				.stream()
+				.sorted(Collections.reverseOrder())
```

Comment:
nice trick to reverse order here, but do we have a test with multiple blank pages interleaved with non-blank pages?

Reply:
This is unrelated to interleaving. The following happens when remoding pages with either asending or descending indexes:
```
Remove with ascending indexes:
Available pages:  0 1 2 3 4 5 6
Pages to remove:  0 1 2
Result (wrong):   1 3 5 6

Remove with descending indexes:
Available pages:  0 1 2 3 4 5 6
Pages to remove:  2 1 0
Result (correct): 3 4 5 6
```

Reply:
I forgot that you only aim to remove leading pages, so all fine. Still, also with non-leading blank pages this would work.